Core/src/OffConverter/OffReader.cpp

```diff
@@ -88,8 +88,47 @@ std::shared_ptr<OffModel> OffReader::readFile(const std::string& filename)
 		}
 		model->addVertices(allVertices);
 
-		//read faces
-		//... still to be added
+		//read faces (special case: after indices of face the color is given in RGB value -> to be considered later on)
+		std::vector<uint32_t> indices;
+		for (int i = 0; i < nrOfFaces; i++)
+		{
+			//which type of face (triangle, quad, ...)
+			std::getline(offFile, line);
+			std::vector<uint32_t> faceVector;
+			std::stringstream lineStream(line);
+
+			int faceType;
+			lineStream >> faceType;
+
+			//read triangle 
+			if (faceType == 3)
+			{
+				lineStream >> faceType >> faceVector[0] >> faceVector[1] >> faceVector[2];
+				for (int j = 0; j < 3; j++)
+					indices.push_back(faceVector[j]);
+			}
+			//read quad
+			else if (faceType == 4)
+			{
+				lineStream >> faceType >> faceVector[0] >> faceVector[1] >> faceVector[2] >> faceVector[3];
+
+				//Convert quads into triangles
+				//first triangle
+				indices.push_back(faceVector[0]);
+				indices.push_back(faceVector[1]);
+				indices.push_back(faceVector[2]);
+				//second triangle
+				indices.push_back(faceVector[2]);
+				indices.push_back(faceVector[3]);
+				indices.push_back(faceVector[0]);
+			}
+			else
+			{
+				offFile.close();
+				throw std::exception("Files that include faces with more than 4 edges are not supported yet.");
```

Comment:
There should be a `NotImplementedYetException` (or something similar) available already.

Reply:
Resolved: 6fad8a0